### PR TITLE
feat: update handshake resistance test

### PR DESCRIPTION
**Pull request by Rqnsom**
- add a set of new subtests

### CODE REVIEW

File: src/tests/resistance/handshake.rs [hunk 1]
```diff
@@ -212,3 +215,185 @@ async fn r001_t5_HANDSHAKE_ws_version() {
...
+    // ZG-RESISTANCE-001
+
+    let gen_cfg = |len| HandshakeCfg {
+        ar_location: Some(gen_huge_string(len)),
+        ..Default::default()
+    };
+
+    // Below tests assert the connection shouldn't be established.
+
+    // NOTE: The node should reject an invalid address.
```
Comment (asmie) [inline src/tests/resistance/handshake.rs:288]: Below tests are commented - maybe leave a TODO to not forget to get back here in the future.

**Review by asmie — APPROVED** [review]
LGTM

**Comment by asmie** [comment]
It can be checked (or changed in the future) if gen_huge_string() should generate garbage strings - like something containing full subset of allowed characters that can be sent like dots, dashes or anything else that could possibly be handled wrong in desired context.
Those tests should also cover cases where headers are below the maximum length but contain garbage characters that makes no sense in such header, just to ensure that header will be processed (as it's valid length).

File: src/tests/resistance/handshake.rs (hunk 1 above)
Comment (Rqnsom) [reply src/tests/resistance/handshake.rs:288]: We'll check these Notes once we go through this with stakeholders. I feel a `TODO` should be used when there is something to do, and we should separate those from these questionable observations we have here

**Pull request merged**